components/tabs/tab-mixin.js

```diff
@@ -0,0 +1,193 @@
+import '../colors/colors.js';
+import { css, html, unsafeCSS } from 'lit';
+import { classMap } from 'lit/directives/class-map.js';
+import { getFocusPseudoClass } from '../../helpers/focus.js';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+
+const keyCodes = {
+	ENTER: 13,
+	SPACE: 32
+};
+
+export const TabMixin = superclass => class extends SkeletonMixin(superclass) {
+	static get properties() {
+		return {
+			selected: { type: Boolean, reflect: true },
+		};
+	}
+
+	static styles = css`
+		:host {
+			box-sizing: border-box;
+			display: inline-block;
+			max-width: 200px;
+			outline: none;
+			position: relative;
+			vertical-align: middle;
+		}
+		.d2l-tab-text {
+			margin: 0.5rem;
+			overflow: hidden;
+			padding: 0.1rem;
+			text-overflow: ellipsis;
+			white-space: nowrap;
+		}
+		:host(:first-child) .d2l-tab-text {
+			margin-inline-end: 0.6rem;
+			margin-inline-start: 0;
+		}
+		.d2l-tab-selected-indicator {
+			border-top: 4px solid var(--d2l-color-celestine);
+			border-top-left-radius: 4px;
+			border-top-right-radius: 4px;
+			bottom: 0;
+			display: none;
+			margin: 1px 0.6rem 0 0.6rem;
+			position: absolute;
+			transition: box-shadow 0.2s;
+			width: calc(100% - 1.2rem);
+		}
+		:host(:first-child) .d2l-tab-selected-indicator {
+			margin-inline-end: 0.6rem;
+			margin-inline-start: 0;
+			width: calc(100% - 0.6rem);
+		}
+		:host(:${unsafeCSS(getFocusPseudoClass())}) > .d2l-tab-text {
+			border-radius: 0.3rem;
+			box-shadow: 0 0 0 2px var(--d2l-color-celestine);
+			color: var(--d2l-color-celestine);
+		}
+		:host([aria-selected="true"]:focus) {
+			text-decoration: none;
+		}
+		:host(:hover) {
+			color: var(--d2l-color-celestine);
+			cursor: pointer;
+		}
+		:host([aria-selected="true"]:hover) {
+			color: inherit;
+			cursor: default;
+		}
+		:host([aria-selected="true"]) .d2l-tab-selected-indicator {
+			display: block;
+		}
+    `;
+
+	constructor() {
+		super();
+		this.ariaSelected = false;
+		this.role = 'tab';
+		this.selected = false;
+		this.tabIndex = -1;
+
+		this.#handleClickBound = this.#handleClick.bind(this);
```

Comment:
Do we need to `bind` these handlers? I _think_ `this` should already be bound to this component for these events. Normally we would only need to `bind` when `this` is something else, like when wiring up to elements outside of this component such as the `window` or `document`.